lib/mixlib/install/generator/bourne/scripts/platform_detection.sh

```diff
@@ -76,12 +76,6 @@ elif test -f "/usr/bin/sw_vers"; then
   platform="mac_os_x"
   # Matching the tab-space with sed is error-prone
   platform_version=`sw_vers | awk '/^ProductVersion:/ { print $2 }' | cut -d. -f1,2`
```

Comment:
While you're in here this whole parsing of sw_vers makes no sense. If you have sw_vers you're on a mac. Making this a complex parsing is what broke things on macOS 11

Reply:
That's where we get the platform version from. If we remove that logic how would we detect the platform version?

My current MBP:
```
➜  sw_vers
ProductName:	Mac OS X
ProductVersion:	10.15.7
BuildVersion:	19H2```

Our Apple Silicon Mac mini:
```
releng@virtmac01 ~ % sw_vers
ProductName:	macOS
ProductVersion:	11.0
BuildVersion:	20A5299w
```

Reply:
Ignore me entirely here. We are actually doing the right thing here. It's our other 2 sets of detection logic for macOS that are still flawed.

Reply:
@tas50 Tell me more...what are the two checks you speak of?

Reply:
Ohai and train. First rule of chef: https://youtu.be/VTUrdizRZyw